### PR TITLE
adding DDS family sku support

### PR DESCRIPTION
**Pull request by jhoreman**
### Which issue this PR addresses:

Addresses requests made in https://issues.redhat.com/browse/ARO-7187 and https://issues.redhat.com/browse/ARO-7816
Fixes

### What this PR does / why we need it:

Provides DDS Family SKU support for control planes

### Test plan for issue:

<!--
How did you test that this PR works?

Local testing and resizing was accepted by the cluster, but will need more validation based on this PR. 

### Is there any documentation that needs to be updated for this PR?
Yes, we need to update our supported SKUs publicly if accepted.

### How do you know this will function as expected in production? 

Resize target is available and successful

### CODE REVIEW

**Review by mociarain — APPROVED** [review]
Nice. Are there MSFT docs we need to update too? A quick google shows we somehow had a release not in the past.
- [Announcement](https://azure.microsoft.com/en-us/updates/new-instance-types-for-azure-red-hat-openshift-are-now-available/)
- [MSFT Notes](https://learn.microsoft.com/en-us/azure/openshift/support-policies-v4#supported-virtual-machine-sizes)

**Review by anshulvermapatel — APPROVED** [review]
LGTM

**Comment by SudoBrendan** [comment]
I think these changes are correct and comprehensive in theory, let's get a full test done before merge (added `hold` until that is done).

File: pkg/api/openshiftcluster.go [hunk 1]
```diff
@@ -445,6 +445,13 @@ const (
 	VMSizeStandardD64asV5 VMSize = "Standard_D64as_v5"
 	VMSizeStandardD96asV5 VMSize = "Standard_D96as_v5"
 
+	VMSizeStandardD4dsV5  VMSize = "Standard_D4ds_v5"
```
[the remark this answers is not in the document]
Comment (jhoreman) [reply pkg/api/openshiftcluster.go:448]: This is a good question. I don't believe we ever expect deviation here but likely have legacy reasons for duplicating these lists in an admin and public context. For what it's worth we intend to leverage the control plane machine set operator in the near future which should no longer look to these lists for validation.

[the remark this answers is not in the document]
Comment (jhoreman) [reply pkg/api/openshiftcluster.go:448]: I'll resolve this for now with the above explanation.

**Comment by jhoreman** [comment]
> I think these changes are correct and comprehensive in theory (we just need a `dsv5` for every `v3` instance type we support), let's get a full master resize test done before merge (added `hold` until that is done).

Given that control plane resize is an SRE driven administrative task only right now, as well as some restrictions on the lower environments I am comfortable merging this and leveraging our new Control Plane Resize GA on a shared test cluster in order to validate.

**Review by tsatam — APPROVED** [review]

**Pull request merged**